docs/2.10.0/docs/concepts/ledger-model/upgrade-model-in-depth.rst

```diff
@@ -2472,3 +2472,595 @@ The static type of the key being looked up is thus ``p-2.0.0:MyKey``. The key of
+requested, then the values in ledger API responses are guaranteed **not** to be
+in normal form.
+
+**Example 1**
+
+Assume a LF 1.17 package called ``example1-1.0.0`` which defines a template
+``T`` and a record ``Record`` in a module called ``Main``.
+
+.. code:: daml
+
+    module Main where
+
+    data Record = Record { ri : Optional Int, rj : Int, rk : Optional Int }
+      deriving (Eq, Show)
+    
+    template T
+      with
+        p : Party
+        i : Optional Int
+        r : Record
+        j : Optional Int
+      where
+        signatory p
+        
+Also assume a ledger that contains a contract of type ``T`` written by
+``example1-1.0.0`` where all the optional fields are set to ``None``.
+
+.. code:: scala
+
+    val createCmd = ledger_api_utils.create(
+      packageIdExample1, 
+      "Main",
+      "T", 
+      Map(
+        "p" -> sandbox.adminParty, 
+        "i" -> None,
+        "r" -> Map("ri" -> None, "rj" -> 1, "rk" -> None),
+        "j" -> None))
+    
+    sandbox.ledger_api.commands.submit(Seq(sandbox.adminParty), Seq(createCmd))
+    
+Then querying the ledger's active contract set in non-verbose mode returns the
+following:
+
+.. code:: scala
+
+    @ sandbox.ledger_api.acs.of_party(sandbox.adminParty, verbose=false) 
+    res15: Seq[com.digitalasset.canton.admin.api.client.commands.LedgerApiTypeWrappers.WrappedCreatedEvent] = List(
+      WrappedCreatedEvent(
+        event = CreatedEvent(
+          ... 
+          createArguments = Some(
+            value = Record(
+              recordId = None,
+              fields = Vector(
+                RecordField(
+                  label = "",
+                  value = Some(value = Value(sum = Party(value = "sandbox::122010fdef685011beecd318f03c9d82bf1e2d45950bdb0fceb3497a112ee17f9476")))
+                ),
+                RecordField(label = "", value = Some(value = Value(sum = Optional(value = Optional(value = None))))),
+                RecordField(
+                  label = "",
+                  value = Some(
+                    value = Value(
+                      sum = Record(
+                        value = Record(
+                          recordId = None,
+                          fields = Vector(
+                            RecordField(label = "", value = Some(value = Value(sum = Optional(value = Optional(value = None))))),
+                            RecordField(label = "", value = Some(value = Value(sum = Int64(value = 1L))))
+                          )
+                        )
+                      )
+                    )
+                  )
+                )
+              )
+            )
+          ),
+          ... 
+        )
+      )
+    )
+
+Note that not only field ``j`` has been omitted from the response, but also
+``rk``. Note also that despite being optional fields of value ``None``, ``i``
+and ``ri`` are present in the response because they are not in trailing
+position.
```

Comment:
```suggestion
Note that not only has field ``j`` been omitted from the response, but also field
``rk``. Note also that despite being optional fields of value ``None``, ``i``
and ``ri`` are present in the response because they are not in trailing
positions.
```